sql/core/src/test/scala/org/apache/spark/sql/NullAwareAntiJoinSQLQueryTestSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.io.File
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+ * End-to-end test cases for subquery SQL queries coverage with
+ * NULL_AWARE_ANTI_JOIN_OPTIMIZE_ENABLED = true.
+ *
+ * Each case is loaded from a file in
+ * "spark/sql/core/src/test/resources/sql-tests/inputs/subquery".
+ * Each case has a golden result file in
+ * "spark/sql/core/src/test/resources/sql-tests/results/subquery".
+ *
+ * To run the entire test suite:
+ * {{{
+ *   build/sbt "sql/test-only *NullAwareAntiJoinSQLQueryTestSuite"
+ * }}}
+ *
+ */
+class NullAwareAntiJoinSQLQueryTestSuite extends SQLQueryTestSuite {
```

Comment:
It's super over-kill to rerun the entire `SQLQueryTestSuite` with NAAJ enabled.

Let's find out the join test files, and use config dimension to test NAAJ. e.g. `join.sql`

Reply:
Actually, very few queries can benefit from NAAJ. Can you find them out and test them via config dimensions?

Reply:
since spark.sql.nullAwareAntiJoin.optimize.enabled is default true for now, is it ok to just delete this case?

Reply:
I will put config dimension in files in sql-tests if there are SQL include "NOT IN"